pydantic/_internal/_generate_schema.py

```diff
@@ -1180,9 +1193,8 @@ def inner_handler(obj: Any) -> CoreSchema:
             get_inner_schema = self._get_wrapped_inner_schema(get_inner_schema, annotation, pydantic_js_functions)
 
         schema = get_inner_schema(source_type)
-        metadata_schema = resolve_original_schema(schema, self.defs.definitions)
```

Comment:
Cleaning up the enum todo let us remove this line which really never should have been here in the first place. If you think about, we expressly _do not_ want to apply the metadata to the original schema.